DllImportGenerator/DllImportGenerator/Marshalling/ICustomNativeTypeMarshallingStrategy.cs

```diff
@@ -632,6 +673,14 @@ public IEnumerable<StatementSyntax> GenerateUnmarshalStatements(TypePositionInfo
                             IdentifierName(ManualTypeMarshallingHelper.SetUnmarshalledCollectionLengthMethodName)))
                     .AddArgumentListArguments(Argument(numElementsExpression)));
             }
+        }
+
+        public IEnumerable<StatementSyntax> GenerateUnmarshalStatements(TypePositionInfo info, StubCodeContext context)
+        {
+            foreach (var statement in GenerateUnmarshallerCollectionInitialization(info, context))
```

Comment:
Is this an intentional contract? We do the current prior to the inner. If so, we should state that here if it creates functional side-effects.